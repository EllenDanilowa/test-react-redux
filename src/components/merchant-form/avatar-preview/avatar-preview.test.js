import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import AvatarPreview from './avatar-preview';

describe('AvatarPreview', () => {
  let alt;
  let element;

  const createElement = ({files = {}, ...props}, isSnapshot = true) => {
    const template = (<AvatarPreview files={files} {...props} />);

    return isSnapshot ?
      renderer.create(template).toJSON() :
      shallow(template);
  };

  beforeEach(() => {
    alt = 'preview alt title';
  });

  it('renders preview when default url is defined', () => {
    const defaultImageUrl = 'url/to/image';

    element = createElement({defaultImageUrl, alt});

    expect(element).toMatchSnapshot();
  });

  it('updates preview if file exists', async () => {
    const setState = jest.fn();
    const files = {
      0: new Blob([''], {type: 'image/png'}),
      length: 1
    };
    const convertedImage = 'data:image/png;base64,';
    jest.spyOn(React, 'useState').mockImplementation((init) => [init, setState]);

    await renderer.act(async () => {
      element = createElement({files, alt});
    });

    expect(setState).toHaveBeenCalledWith(convertedImage);
  });
});
