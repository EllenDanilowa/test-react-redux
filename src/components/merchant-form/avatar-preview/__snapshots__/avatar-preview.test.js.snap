// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvatarPreview renders preview when default url is defined 1`] = `
<div
  className="sc-AxjAm jpchXk"
>
  
  <img
    alt="preview alt title"
    className="sc-AxirZ cUVYWJ"
    src="url/to/image"
  />
</div>
`;
